solarforecastarbiter/validation/quality_mapping.py

```diff
@@ -0,0 +1,133 @@
+"""
+Define constant mappings between bit-mask values and understandable quality
+flags
+"""
+from functools import wraps
+
+
+# The quality_flag field in MySQL is currently limited to 1 << 15;
+# fields beyond 1 << 15 will require a change in the MySQL datatype
+# for the quality_flag column. The mapping from description to bitmask
+# is versioned so that future addtions or removals are backwards compatible
+# without rerunning the validation on all data.
+# DO NOT MODIFY THE VALUES OF THE _BITMASK_DESCRIPTION_DICT instead,
+# add a increment the key and add a new value tuple. The VERSION
+# IDENTIFIER 0 - 2 must remain in their current positions. Versions 7
+# and up will require another identifier bit to be determined at that
+# time.  The version identifier also serves to mark data in the
+# database as validated. The tuples are (description, bit mask, flag class)
+_BITMASK_DESCRIPTION_DICT = {1: (
+    # start with 1 to distinguish validated vs not in DB
+    ('OK', 0, ''),
+    ('USER FLAGGED', 1 << 0, 'bad'),
+    ('VERSION IDENTIFIER 0', 1 << 1, ''),
+    ('VERSION IDENTIFIER 1', 1 << 2, ''),
+    ('VERSION IDENTIFIER 2', 1 << 3, ''),
+    ('NIGHTTIME', 1 << 4, 'informational'),
+    ('CLOUDY', 1 << 5, 'informational'),
+    ('SHADED', 1 << 6, 'informational'),
+    ('UNEVEN FREQUENCY', 1 << 7, 'bad'),
+    ('LIMITS EXCEEDED', 1 << 8, 'bad'),
+    ('CLEARSKY EXCEEDED', 1 << 9, 'bad'),
+    ('STALE VALUES', 1 << 10, 'bad'),
+    ('INTERPOLATED VALUES', 1 << 11, 'bad'),
+    ('CLIPPED VALUES', 1 << 12, 'bad'),
+    ('INCONSISTENT IRRADIANCE COMPONENTS', 1 << 13, 'bad'),
+    ('RESERVED 0', 1 << 14, ''),  # available for new flag
+    ('RESERVED 1', 1 << 15, ''),  # available for new flag
+    )
+}
+
+
+# should never change unless another VERSION IDENTIFIER is required
+VERSION_MASK = 0b1110
+LATEST_VERSION = max(_BITMASK_DESCRIPTION_DICT.keys())
+LATEST_BITMASK_DESCRIPTION = _BITMASK_DESCRIPTION_DICT[LATEST_VERSION]
+DESCRIPTION_MASK_MAPPING = {v[0]: v[1] for v in LATEST_BITMASK_DESCRIPTION}
+MASK_DESCRIPTION_MAPPING = {v[1]: v[0] for v in LATEST_BITMASK_DESCRIPTION}
+LATEST_VERSION_FLAG = (
+    LATEST_VERSION * DESCRIPTION_MASK_MAPPING['VERSION IDENTIFIER 0'])
+
+
+def convert_bool_flags_to_flag_mask(flags, flag_description, invert):
+    if flags is None:
+        return None
+    if invert:
+        bool_flags = ~(flags.astype(bool))
+    else:
+        bool_flags = flags.astype(bool)
+    return ((bool_flags * DESCRIPTION_MASK_MAPPING[flag_description])
+            | LATEST_VERSION_FLAG)
+
+
+def mask_flags(flag_description, invert=True):
+    """
+    Decorator to convert boolean flags to the integer of flag_description.
+    Will only apply if the 'return_bool' keyword is pass to the decorated
```

Comment:
It checks if the ``return_bool`` kwarg is present. If ``return_bool = False``, then instead of returning a boolean series, return an integer bitmask with the appropriate value. 

I like the decorator because it clearly identifies how each function in validator.py will be represented in ``quality_flag``. The whole ``return_bool`` kwarg is primarily internal  so we don't need to keep track of function/series names. I think if I separate the validator calls in ``tasks.py`` instead of immediately combining them, debugging will be just as easy as with a boolean series.

Reply:
not sure about rendering yet